packages/core/contracts/financial-templates/expiring-multiparty/PricelessPositionManager.sol

```diff
@@ -790,16 +796,31 @@ contract PricelessPositionManager is FeePayer {
 
     // Requests a price for transformed `priceIdentifier` at `requestedTime` from the Oracle.
     function _requestOraclePrice(uint256 requestedTime) internal {
-        OracleInterface oracle = _getOracle();
-        oracle.requestPrice(_transformPriceIdentifier(requestedTime), requestedTime);
+        OptimisticOracleInterface oracle = _getOptimisticOracle();
+        oracle.requestPrice(
+            _transformPriceIdentifier(requestedTime),
+            requestedTime,
+            _getAncillaryData(),
+            collateralCurrency,
+            0
+        );
     }
 
     // Fetches a resolved Oracle price from the Oracle. Reverts if the Oracle hasn't resolved for this request.
-    function _getOraclePrice(uint256 requestedTime) internal view returns (FixedPoint.Unsigned memory) {
+    function _getOraclePrice(uint256 requestedTime) internal returns (FixedPoint.Unsigned memory) {
```

Comment:
this `view` modifier needed to be removed as the optimistic oracle's `hasPrice` is state modifying.